compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -352,7 +352,8 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
 
         let ty_span = match rcvr_ty.kind() {
             ty::Param(param_type) => {
-                Some(param_type.span_from_generics(self.tcx, self.body_id.owner.to_def_id()))
+                let body_hir_id = self.tcx.hir().local_def_id_to_hir_id(self.body_id);
+                Some(param_type.span_from_generics(self.tcx, body_hir_id.owner.to_def_id()))
```

Comment:
```suggestion
                Some(param_type.span_from_generics(self.tcx, self.body_id))
```

Reply:
```
   Compiling rustc_hir_typeck v0.1.0 (/home/vincent/Github/rust/compiler/rustc_hir_typeck)
error[E0308]: mismatched types
    --> compiler/rustc_hir_typeck/src/method/suggest.rs:355:62
     |
355  |                 Some(param_type.span_from_generics(self.tcx, self.body_id))
     |                                 ------------------           ^^^^^^^^^^^^ expected struct `rustc_span::def_id::DefId`, found struct `LocalDefId`
     |                                 |
     |                                 arguments to this function are incorrect
     |
note: associated function defined here
    --> /home/vincent/Github/rust/compiler/rustc_middle/src/ty/sty.rs:1316:12
     |
1316 |     pub fn span_from_generics(&self, tcx: TyCtxt<'tcx>, item_with_generics: DefId) -> Span {
     |            ^^^^^^^^^^^^^^^^^^
help: call `Into::into` on this expression to convert `LocalDefId` into `rustc_span::def_id::DefId`
     |
355  |                 Some(param_type.span_from_generics(self.tcx, self.body_id.into()))

```

I was not able to get what is the say way to pass from LocalDefId to a DefId

Reply:
There's a `to_def_id` method on `LocalDefId`